packages/@aws-cdk/aws-pipes-targets-alpha/test/integ.api-destination.ts

```diff
@@ -0,0 +1,106 @@
+import { IPipe, ISource, Pipe, SourceConfig } from '@aws-cdk/aws-pipes-alpha';
+import { ExpectedResult, IntegTest } from '@aws-cdk/integ-tests-alpha';
+import * as cdk from 'aws-cdk-lib';
+import * as apigwv2 from 'aws-cdk-lib/aws-apigatewayv2';
+import { HttpLambdaIntegration } from 'aws-cdk-lib/aws-apigatewayv2-integrations';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import { ApiDestinationTarget } from '../lib/api-destination';
+
+const app = new cdk.App();
+const stack = new cdk.Stack(app, 'aws-cdk-pipes-targets-api-dest');
+const sourceQueue = new cdk.aws_sqs.Queue(stack, 'SourceQueue');
+
+/*
+ * This integration test creates an API Gateway HTTP API in order to
+ * validate that the pipe executes properly. The flow is as follows:
+ *
+ * SQS (pipe source) --> EventBridge API destination (pipe target)
+ * --> API Gateway HTTP API --> Lambda function
+ */
+
+class TestSource implements ISource {
+  sourceArn: string;
+  sourceParameters = undefined;
+  constructor(private readonly queue: cdk.aws_sqs.Queue) {
+    this.queue = queue;
+    this.sourceArn = queue.queueArn;
+  }
+  bind(_pipe: IPipe): SourceConfig {
+    return {
+      sourceParameters: this.sourceParameters,
+    };
+  }
+  grantRead(pipeRole: cdk.aws_iam.IRole): void {
+    this.queue.grantConsumeMessages(pipeRole);
+  }
+}
```

Comment:
why do you create this Source class, and not to use the `SqsSource` available in the pipes-sources module

Reply:
I was reusing the pattern in other integration tests. I think it's so we don't need to import `SqsSource` and instead use the interface?

[1] https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-pipes-targets-alpha/test/integ.stepfunctions.ts#L39
[2] https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-pipes-targets-alpha/test/integ.sqs.ts#L12
[3] https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-pipes-targets-alpha/test/integ.lambda.ts#L46

Reply:
IMO, in integration test cases we should use CDK Apis the same way the customer will use them, so I prefer we also change the other integration test cases, but this is out of this PR scope. But at least for this PR, let's use the SqsSource

Reply:
Fixed in https://github.com/aws/aws-cdk/pull/30756/commits/a4f52b2454ae2be4666288b20d4b53ced0bedfe3